certora/specs/SocialRecoveryModule.spec

```diff
@@ -1,26 +1,215 @@
+// - the linked list integrity remains,
+// - and no other account (guardian or not) is affected.
+rule revokeGuardiansWorksAsExpected(env e, address guardian, address prevGuardian, uint256 threshold, address otherAccount) {
+    address nextGuardian = guardianStorageContract.entries[safeContract].guardians[guardian];
+
+    require guardian != otherAccount;
+    bool otherAccountIsGuardian = currentContract.isGuardian(safeContract, otherAccount);
+
+    uint256 currentGuardiansCount = guardianStorageContract.entries[safeContract].count;
+    require currentGuardiansCount > 0;
+
+    requireGuardiansLinkedListIntegrity();
+    require guardianStorageContract.entries[safeContract].guardians[guardian] != guardian;
+
+    currentContract.revokeGuardianWithThreshold(e, safeContract, prevGuardian, guardian, threshold);
 
+    assert safeContract.isModuleEnabled(currentContract);
+    assert e.msg.sender == safeContract;
+    assert !currentContract.isGuardian(safeContract, guardian);
+    assert guardianStorageContract.entries[safeContract].guardians[prevGuardian] == nextGuardian;
+    assert otherAccountIsGuardian == currentContract.isGuardian(safeContract, otherAccount);
+    assert currentGuardiansCount - 1 == to_mathint(guardianStorageContract.entries[safeContract].count);
+}
+
+// This integrity rule verifies that the guardian can always be revoked considering ideal conditions.
+// Some of the conditions are:
+// - No value should be sent with the transaction.
+// - Guardian should not be zero address.
+// - New threshold should be greater than 0.
+// - The guardian count should be greater than the new threshold.
+// - The address should be a guardian.
+// - Safe Contract should be the sender of the transaction.
+rule guardianCanAlwaysBeRevoked(env e, address guardian, address prevGuardian, uint256 threshold) {
     requireSocialRecoveryModuleEnabled();
 
-    safeContract.disableModule@withrevert(e, prevModule, currentContract);
+    require e.msg.value == 0;
+    require guardian != 0;
+    require threshold > 0;
+    require guardianStorageContract.countGuardians(safeContract) > threshold;
+    require currentContract.isGuardian(safeContract, guardian);
+    require guardianStorageContract.entries[safeContract].guardians[guardian] != guardian;
+    requireGuardiansLinkedListIntegrity();
+
+    address nextGuardian = guardianStorageContract.entries[safeContract].guardians[guardian];
+    require guardianStorageContract.entries[safeContract].guardians[prevGuardian] == guardian;
+
+    require e.msg.sender == safeContract;
+    currentContract.revokeGuardianWithThreshold@withrevert(e, safeContract, prevGuardian, guardian, threshold);
     bool isReverted = lastReverted;
 
-    assert !isReverted => !safeContract.isModuleEnabled(currentContract);
+    assert !isReverted &&
+        guardianStorageContract.entries[safeContract].guardians[prevGuardian] == nextGuardian &&
+        !currentContract.isGuardian(safeContract, guardian);
```

Comment:
We should also check that revoking guardians updates the `count` right?

Reply:
Done 👍🏾